router/server.go

```diff
@@ -46,7 +46,7 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int, g http.
 	// 	Handler(http.StripPrefix("/provider/", http.FileServer(http.Dir("../provider-ui/out/")))).
 	// 	Methods("GET")
 
-	gMux.Handle("/api/system/sync", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.SessionSyncHandler)))).
+	gMux.Handle("/api/system/sync", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.KubernetesMiddleware(h.MesheryControllersMiddleware(h.SessionSyncHandler)))))).
```

Comment:
This middleware is added to all the handlers that use the contexts. If you find that there are more handlers to which this middleware has to be added, please let me know.

Reply:
Hmm I'll have to see. But we should note this somewhere so that in future if any new endpoint or existing one tries to use the k8scontexts, they will have to add this middleware.

Reply:
Yes, the point is to have more control and trasnparency.